src/modules/flow/si114x/si114x.c

```diff
@@ -0,0 +1,370 @@
+ *   * Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in
+ *     the documentation and/or other materials provided with the
+ *     distribution.
+ *   * Neither the name of Intel Corporation nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <errno.h>
+#include <limits.h>
+#include <regex.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "si114x-gen.h"
+#include "sol-flow.h"
+#include "sol-mainloop.h"
+#include "sol-util.h"
+#include "sol-i2c.h"
+#include "sol-types.h"
+#include "sol-flow-internal.h"
+
+/*
+ * SI114X registers
+ */
+enum _SI114X_REG {
+    REG_PART_ID       = 0x00,
+    REG_REV_ID        = 0x01,
+    REG_SEQ_ID        = 0x02,
+    REG_INT_CFG       = 0x03,
+    REG_IRQ_ENABLE    = 0x04,
+
+    // these two are not documented in the datasheet,
+    // but are mentioned there, as well as in the
+    // Adafruit example
+    REG_IRQ_MODE1     = 0x05,
+    REG_IRQ_MODE2     = 0x06,
+
+    REG_HW_KEY        = 0x07,
+    REG_MEAS_RATE0    = 0x08,
+    REG_MEAS_RATE1    = 0x09,
+
+    REG_PS_LED21      = 0x0f,
+    REG_PS_LED3       = 0x10,
+
+    REG_UCOEF0        = 0x13,
+    REG_UCOEF1        = 0x14,
+    REG_UCOEF2        = 0x15,
+    REG_UCOEF3        = 0x16,
+    REG_PARAM_WR      = 0x17,
+    REG_COMMAND       = 0x18,
+
+    REG_RESPONSE      = 0x20,
+    REG_IRQ_STATUS    = 0x21,
+    REG_ALS_VIS_DATA0 = 0x22,
+    REG_ALS_VIS_DATA1 = 0x23,
+    REG_ALS_IR_DATA0  = 0x24,
+    REG_ALS_IR_DATA1  = 0x25,
+    REG_PS1_DATA0     = 0x26,
+    REG_PS1_DATA1     = 0x27,
+    REG_PS2_DATA0     = 0x28,
+    REG_PS2_DATA1     = 0x29,
+    REG_PS3_DATA0     = 0x2a,
+    REG_PS3_DATA1     = 0x2b,
+    REG_AUX_UVINDEX0  = 0x2c,
+    REG_AUX_UVINDEX1  = 0x2d,
+    REG_PARAM_READ    = 0x2e,
+
+    REG_CHIP_STAT     = 0x30,
+
+    REG_ANA_IN_KEY0   = 0x3b,
+    REG_ANA_IN_KEY1   = 0x3c,
+    REG_ANA_IN_KEY2   = 0x3d,
+    REG_ANA_IN_KEY3   = 0x3e
+};
+typedef enum _SI114X_REG SI114X_REG_T;
```

Comment:
Also we're not spacing enum value entries like that (we don't enforce the other way yet, but...)